Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp

```diff
@@ -361,7 +361,7 @@ void WebAssemblyModuleRecord::initializeImports(JSGlobalObject* globalObject, JS
                 if (!value.inherits<JSWebAssemblyGlobal>())
                     return exception(createJSWebAssemblyLinkError(globalObject, vm, importFailMessage(import, "imported global"_s, "must be a WebAssembly.Global object since it is mutable"_s)));
                 JSWebAssemblyGlobal* globalValue = jsCast<JSWebAssemblyGlobal*>(value);
-                if (!isSubtype(globalValue->global()->type(), global.type))
+                if (!isSubtype(globalValue->global()->type(), global.type) || !isSubtype(global.type, globalValue->global()->type()))
```

Comment:
Do we also need the direction check for `!Wasm::isSubtype(globalValue->global()->type(), global.type)` when `value.inherits<JSWebAssemblyGlobal>()` is true in the code above? Maybe we should extract the check as `isSameType` or something similar.

Reply:
It's checking the JSWebAssemblyGlobal storage type is a subtype of the expected type (and visa versa).

Reply:
We don't, because the above block is for immutable globals. In the case that we import a global as a supertype of its original type, we'd also need to be able to mutate it in order to give the original global an illegal value. If we can't mutate it, we just care that our imported version can represent all possible values of the original, which is a single subtype check.